x-pack/plugins/observability/common/constants.ts

```diff
@@ -10,7 +10,7 @@ import { AlertConsumers } from '@kbn/rule-data-utils';
 import type { ValidFeatureId } from '@kbn/rule-data-utils';
 
 export const SLO_BURN_RATE_RULE_TYPE_ID = 'slo.rules.burnRate';
-export const OBSERVABILITY_THRESHOLD_RULE_TYPE_ID = 'observability.rules.threshold';
```

Comment:
I believe you would need to have a migration to change the rule type id.  All the existing rules of that type would need to be changed, for instance.  Perhaps some other things, not sure, as I'm not sure we've made a change like this recently.

ZDT throws in more complications, and presumably the deployment would not be able to be "rolled back" after the migration occurs.

I think you're going to have to keep the original rule type string - but if it really needs to change, we can dig in and see if it's possible - I think the "can't roll back" is the killer here though.

Reply:
@pmuellr, does the fact that it's behind a feature flag and in a tech preview stage (no users yet) eliminate the concerns of migration or leaving old rules behind?

Reply:
yes, perhaps in that case it's fine.  Team is discussing right now ...

Reply:
If you're sure no customers will have the rule with the old rule type loaded, you should be good.   

You should try doing a "migration test", where you run  main ES + Kibana, using the es param `-E path.data=path-to-es-data`, add the rule with the old rule type, let it run, then shut everything down, run your PR ES + Kibana with the same `-E` param for es, so it will "migrate" the deployment.  See how bad it is if someone DOES leave one in, like in an oblt cluster or similar.

We also have some "block lists" for task and/or rule types, that are hard-coded somewhere, if we need to exclude the old ones.

Reply:
@pmuellr, thank you for sharing this. I wasn't aware of such a process!